react.gradle

```diff
@@ -26,7 +26,7 @@ def detectEntryFile(config) {
  */
 def detectCliPath(config) {
     if (config.cliPath) {
-        return "${projectDir}/${config.cliPath}"
+        return (config.cliPath.startsWith("/")) ? config.cliPath : "${projectDir}/${config.cliPath}"
```

Comment:
Thanks for this. However I would like to keep `react.gradle` aligned with our Gradle Plugin code:

https://github.com/facebook/react-native/blob/b3c69e85d78fb221f271d0eb608c7bb274ecbd6d/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/PathUtils.kt#L88-L92

So let's maybe check if the file exists instead of checking if `.startsWith("/")` here?

Reply:
Thanks for the feedback, tried to do it as minimal as possible.
What do you think about doing a slightly bigger change in that spirit?

The following would use `.exists()` check, and also show an explicit error when an erroneous `cliPath` is provided, as opposed to having to dig through logs and code for the info.

```groovy
def detectCliPath(config) {
    if (config.cliPath) {
        if (new File(config.cliPath).exists()) {
            return config.cliPath
        }
        if (new File("${projectDir}/${config.cliPath}").exists()) {
            return "${projectDir}/${config.cliPath}"
        }
    } else if (new File("${projectDir}/../../node_modules/react-native/cli.js").exists()) {
        return "${projectDir}/../../node_modules/react-native/cli.js"
    }
    throw new Exception("Couldn't determine CLI location. " +
             "Please set `project.ext.react.cliPath` to the path of the react-native cli.js");
}
```

Reply:
> What do you think about doing a slightly bigger change in that spirit?

That's a change in the right step. However I'd try to keep the two `detectCliPath` aligned if possible. How do you feel about implementing the same login from `PathUtils.kt` inside `react.gradle`?

Reply:
@cortinico Can you please review my PR as well, I have #32956

Reply:
I guess I'll make changes to `PathUtils` as well, to include absolute pathing, and `.exists` checks

Reply:
That's also great. Please udpate PathUtilsTest.kt as well 👍